### ISSUE TITLE
Moving iframe in DOM - can we re-render?

### ISSUE THREAD

**Issue by joekur** [tiff/wysihtml5#148]
I noticed while trying to use jquery drag and drop on something that contained a wysihtml5 iframe that when you move the iframe within the dom, it loses all its functionality (and contains nothing inside). Is there someway to circulate this issue or "restart" the iframe?

**Comment by joekur** [comment]
I believe the issue is that the iframe reloads. But correct me if I'm wrong.

**Comment by tiff** [comment]
I asume that this is only happening in Firefox, right?
See https://bugzilla.mozilla.org/show_bug.cgi?id=254144

**Comment by joekur** [comment]
No, I initially found the issue using chrome. But firefox has the issue too.

**Comment by perymimon** [comment]
I have researched this thing for a different purpose. And definitely has rebuilding all the window within the iframe.
But the previous window still exists.
Then note that if you hold a refrence to that iframe window before moving. After transition the reference become irrelevant

**Comment by mateusmaso** [comment]
+1 for restarting wysihtml5 when iframe is reloaded. Im trying to use jQuery sortable also but then I get blank state when finish moving the editor (chrome). Is there any workaround for it? As I see, there is no clean way to stop the default behavior when iframes change the DOM tree.

**Comment by joekur** [comment]
Mateusmaso, I had the same situation. My current workaround is to completely re-render the textarea and re-initialize wysihtml5. Quick fix because I had it set up with a backbonejs view already, but certainly not an ideal solution.

**Comment by bjoernm** [comment]
+1 for restarting wysihtml5 when iframe is reloaded. Having the same problem as Mateusmaso. I am using jQuery sortable to move a wysithml5 editor and after moving it freezes (Chrome+FF). Joekur's workaround seems to be the only possibility right now.

**Comment by parterburn** [comment]
Same issue - jQuery sortable + wysihtml5 = bad news.

``` javascript
  $( "#logosSortable" ).sortable({
      stop: function(event, ui) {

        //WYSIHTML5 doesn't play nicely with Sortable so we reinitialize when it stops dragging
        $('.wysihtml5-sandbox').remove();
        $('.wysiwyg').css('display','block');
        $('.wysiwyg').wysihtml5();

      }
  });
```

**Comment by bradt** [comment]
I'm seeing the same thing on my end, but I'm not using sortable. If you detach the iframe from the DOM using jQuery `.detach()` and reattach it, it loses its functionality. Haven't found a workaround yet.

**Comment by adamburmister** [comment]
Any work around found for this yet? I'm getting it with sortables.

**Comment by joekur** [comment]
The only way is to re-initialize everything after moving the iframe, as parterburn showed. His example code removes the iframe, redisplays the textarea and then initializes wysihtml5 again.

I can imagine wysihtml5 having a `reload()` function that could be called manually or possibly attached to the iframe's load event (which may require some additional logic so it's not called on a normal load).

**Comment by briansw** [comment]
I had some issues with the recommendations above, but here's how I solved it by completely removing the iframes from the DOM and then reinitializing the textareas again. Maybe someone will find it useful

```
var textareas = [];
set_text_areas();
function set_text_areas() {
    $('.wysihtml5-sandbox').remove();
    $('.textarea').css('display','block');
    textareas.length = 0;
    $('.textarea').each(function(x) {
        var id       = $(this).data('id'),
            textarea = $(this).attr('id'),
            toolbar  = 'textarea-toolbar-' + id;

        textareas[x] = new wysihtml5.Editor(textarea, {
            toolbar:       toolbar,
            parserRules:   wysihtml5ParserRules,
            useLineBreaks: true,
            autoLink:      true
        });
    });
}

$('#sortable').sortable({
    stop: function(event, ui) {
        set_text_areas();
    }
});
```
